src/data-sources/plugins/GraphQL.ts

```diff
@@ -0,0 +1,74 @@
+import * as request from 'xhr-request';
+
+import { DataSourcePlugin, IOptions } from './DataSourcePlugin';
+import GraphQLConnection from '../connections/graphql';
+import { DataSourceConnector } from '../DataSourceConnector';
+
+let connectionType = new GraphQLConnection();
+
+interface IGraphQLParams {
+  query: string;
+  variables: Object;
+}
+
+export default class GraphQL extends DataSourcePlugin<IGraphQLParams> {
+  type = 'GraphQL';
+  defaultProperty = 'data';
+  connectionType = connectionType.type;
+
+  constructor(options: IOptions<IGraphQLParams>, connections: IDict<IStringDictionary>) {
+    super(options, connections);
+    this.validateParams(this._props.params);
+  }
+
+  updateDependencies(dependencies: any) {
+    // Ensure dependencies exist
+    const isAnyDependencyMissing = Object.keys(this.getDependencies()).some(key => dependencies[key] == null);
+    if (isAnyDependencyMissing) {
+      return dispatch => dispatch();
+    }
+
+    // Validate connection
+    const connection = this.getConnection();
+    const { serviceUrl } = connection;
+    if (!connection || !serviceUrl) {
+      return dispatch => dispatch();
+    }
+
+    const params = this.getParams();
+    const query = params.query;
+    const variables = dependencies['variables'] || params.variables;
+
+    return dispatch => {
+      request('/graphql/query', {
+        method: 'POST',
+        json: true,
+        body: {
+          serviceUrl: serviceUrl,
+          query: query,
+          variables: variables
+        }
+      },      (err, json) => {
+        const error = err || (json['errors'] && json['errors'][0]);
+        if (error || json['errors']) {
+          console.log(error);
+          return this.failure(error);
+        }
+
+        return dispatch(json);
+      });
+    };
+  }
+
+  updateSelectedValues(dependencies: IDictionary, selectedValues: any) {
+    if (Array.isArray(selectedValues)) {
+      return Object.assign(dependencies, { 'selectedValues': selectedValues });
```

Comment:
This is copied from other `updateSelectedValues` implementations (e.g. [Sample.ts](https://github.com/CatalystCode/ibex-dashboard/blob/d89db0cbf2184d9efaa82c48f0a13904a3ef251b/src/data-sources/plugins/Sample.ts#L41-L47), [Azure.ts](https://github.com/CatalystCode/ibex-dashboard/blob/d89db0cbf2184d9efaa82c48f0a13904a3ef251b/src/data-sources/plugins/Azure.ts#L97-L103)) so I'd say that we should clean this up in bulk in the future.